homeassistant/components/group/fan.py

```diff
@@ -288,8 +318,26 @@ def async_update_group_state(self) -> None:
         self._set_attr_most_frequent(
             "_oscillating", SUPPORT_OSCILLATE, ATTR_OSCILLATING
         )
+
         self._set_attr_most_frequent("_direction", SUPPORT_DIRECTION, ATTR_DIRECTION)
 
+        all_supported_preset_modes = list(
+            find_state_attributes(on_states, ATTR_PRESET_MODES)
+        )
+
+        if all_supported_preset_modes and attribute_equal(on_states, ATTR_PRESET_MODES):
+            self._preset_modes = all_supported_preset_modes[0]
+        else:
+            self._preset_modes = None
+
```

Comment:
We need to ensure the list of preset modes never changes after the `started` event since this will break anything that looks at them only at startup.

It looks like already handle that case in the `GroupEntity` base class which calls `async_update_group_state`

Untested, but something like this should do the trick

```python3
diff --git a/homeassistant/components/group/fan.py b/homeassistant/components/group/fan.py
index a639248fdc..8c501a1543 100644
--- a/homeassistant/components/group/fan.py
+++ b/homeassistant/components/group/fan.py
@@ -116,6 +116,7 @@ class FanGroup(GroupEntity, FanEntity):
         self._oscillating = None
         self._direction = None
         self._preset_mode = None
+        self._built_preset_modes = False
         self._preset_modes = None
         self._supported_features = 0
         self._speed_count = 100
@@ -287,6 +288,18 @@ class FanGroup(GroupEntity, FanEntity):
         setattr(self, attr, most_frequent_attribute(states, entity_attr))
         self._attr_assumed_state |= not attribute_equal(states, entity_attr)
 
+    @callback
+    def _async_build_preset_modes(self, on_states: list[State]) -> None:
+        """Build the preset mode list once."""
+        all_supported_preset_modes = list(
+            find_state_attributes(on_states, ATTR_PRESET_MODES)
+        )
+        if all_supported_preset_modes and attribute_equal(on_states, ATTR_PRESET_MODES):
+            self._preset_modes = all_supported_preset_modes[0]
+        else:
+            self._preset_modes = None
+        self._built_preset_modes = True
+
     @callback
     def async_update_group_state(self) -> None:
         """Update state and attributes."""
@@ -321,19 +334,14 @@ class FanGroup(GroupEntity, FanEntity):
 
         self._set_attr_most_frequent("_direction", SUPPORT_DIRECTION, ATTR_DIRECTION)
 
-        all_supported_preset_modes = list(
-            find_state_attributes(on_states, ATTR_PRESET_MODES)
-        )
-
-        if all_supported_preset_modes and attribute_equal(on_states, ATTR_PRESET_MODES):
-            self._preset_modes = all_supported_preset_modes[0]
-        else:
-            self._preset_modes = None
+        if not self._built_preset_modes:
+            self._async_build_preset_modes(on_states)
 
-        if attribute_equal(on_states, ATTR_PRESET_MODE):
-            self._set_attr_most_frequent(
-                "_preset_mode", SUPPORT_PRESET_MODE, ATTR_PRESET_MODE
-            )
+        if (
+            attribute_equal(on_states, ATTR_PRESET_MODE)
+            and on_states[0].attributes[ATTR_PRESET_MODE] in self._preset_modes
+        ):
+            self._preset_mode = on_states[0].attributes[ATTR_PRESET_MODE]
         else:
             self._attr_assumed_state = True
             self._preset_mode = None

```